### PR TITLE
executor: remove Next function for NestedLoopApplyExec

### PR DESCRIPTION
**Pull request by yangwenmai**

### CODE REVIEW

**Comment by zz-jason** [comment]
```sh
➜ unused github.com/pingcap/tidb/executor
executor/executor.go:900:2: field asName is unused (U1000)
executor/join.go:516:31: func (*NestedLoopApplyExec).fetchOuterRow is unused (U1000)
executor/join.go:570:31: func (*NestedLoopApplyExec).prepare is unused (U1000)
executor/join.go:626:31: func (*NestedLoopApplyExec).doJoin is unused (U1000)
executor/show.go:61:2: field fetched is unused (U1000)
```
@yangwenmai please remove the unused functions.

**Comment by XuHuaiyu** [comment]
Please fix ci. @yangwenmai 
TestNestedLoopApply uses NestedLoopApply.Next(), please change it to NestedLoopApply.NextChunk()

**Comment by yangwenmai** [comment]
When I fix ci error, I have a problem.

In `executor/pkg_test.go#L91`

```
join.innerChunk = innerExec.newChunk()

row, err := join.Next(ctx)
c.Check(err, IsNil)
c.Check(row, NotNil)

c.Check(fmt.Sprintf("%v %v", row[0].GetValue(), row[1].GetValue()), Equals, "1 1")
```

change into:

```
join.innerChunk = innerExec.newChunk()
join.outerChunk = outerExec.newChunk()

chk := join.newChunk()
err := join.NextChunk(ctx, chk)
row := chk.GetRow(1)
c.Check(err, IsNil)
c.Check(row, NotNil)

fields := make([]*types.FieldType, 2)
fields[0] = types.NewFieldType(mysql.TypeLong)
fields[1] = types.NewFieldType(mysql.TypeLong)
cols := row.GetDatumRow(fields)

c.Check(fmt.Sprintf("%v %v", cols[0].GetValue(), cols[1].GetValue()), Equals, "1 1")
```

The error message `index out of range` is because cols is an empty array.

**Comment by XuHuaiyu** [comment]
You need to implement the NextChunk function for MockExec(pkg_test.go#L22) first.
@yangwenmai

**Comment by XuHuaiyu** [comment]
Please fix go fmt error.

**Comment by yangwenmai** [comment]
Maybe a go fmt error caused when I try to resolve some conflicting code on GitHub.

**Comment by XuHuaiyu** [comment]
you can run `go fmt ./...` at directory `executor`.

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
PTAL @coocood @zz-jason

File: executor/pkg_test.go [hunk 1]
```diff
@@ -41,6 +41,23 @@ func (m *MockExec) Next(ctx context.Context) (Row, error) {
 	return r, nil
 }
 
+func (m *MockExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	rt := m.retTypes()
+	for m.curRowIdx < len(m.Rows) {
```
Comment (zz-jason) [inline executor/pkg_test.go]:
```go
for ; m.curRowIdx < len(m.Rows) && chk.NumRows() < m.maxChunkSize; m.curRowIdx++ {
    ...
}
```

File: executor/pkg_test.go [hunk 2]
```diff
@@ -41,6 +41,23 @@ func (m *MockExec) Next(ctx context.Context) (Row, error) {
 	return r, nil
 }
 
+func (m *MockExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	rt := m.retTypes()
```
Comment (zz-jason) [inline executor/pkg_test.go]: `rt` is too simple, how about `colTypes` ?

File: executor/pkg_test.go [hunk 3]
```diff
@@ -90,27 +112,20 @@ func (s *pkgTestSuite) TestNestedLoopApply(c *C) {
...
-	c.Check(err, IsNil)
-	c.Check(row, NotNil)
-	c.Check(fmt.Sprintf("%v %v", row[0].GetValue(), row[1].GetValue()), Equals, "5 5")
-	row, err = join.Next(ctx)
-	c.Check(err, IsNil)
-	c.Check(row, IsNil)
+	join.outerChunk = outerExec.newChunk()
+	joinChk := join.newChunk()
+	it := chunk.NewIterator4Chunk(joinChk)
+	for i := 1; ; {
```
Comment (zz-jason) [inline executor/pkg_test.go]: how about `s/i/rowIdx/`

File: executor/pkg_test.go [hunk 4]
```diff
@@ -41,6 +41,23 @@ func (m *MockExec) Next(ctx context.Context) (Row, error) {
...
 }
 
+func (m *MockExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	rt := m.retTypes()
+	for m.curRowIdx < len(m.Rows) {
+		curRow := m.Rows[m.curRowIdx]
+		for i := 0; i < len(curRow); i++ {
+			curDatum := curRow.GetDatum(i, rt[i])
+			chk.AppendDatum(i, &curDatum)
```
Comment (zz-jason) [inline executor/pkg_test.go]:
```go
chk.AppendDatum(i, &m.Rows[m.curRowIdx].GetDatum(i, rt[i]))
```

Comment (yangwenmai) [reply executor/pkg_test.go]: cannot take the address of `m.Rows[m.curRowIdx].GetDatum(i, rt[i])`.

https://stackoverflow.com/questions/10535743/address-of-a-temporary-in-go
https://golang.org/ref/spec#Address_operators

**Pull request merged**